Tests/NIOPosixTests/AsyncChannelBootstrapTests.swift

```diff
@@ -470,6 +497,69 @@ final class AsyncChannelBootstrapTests: XCTestCase {
         }
     }
 
+    // MARK: Datagram Bootstrap
+
+    func testDatagramBootstrap_withAsyncChannel_andHostPort() throws {
+        XCTAsyncTest {
+            let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 3)
+            let serverChannel = try await self.makeUDPServerChannel(eventLoopGroup: eventLoopGroup)
+            let clientChannel = try await self.makeUDPClientChannel(
+                eventLoopGroup: eventLoopGroup,
+                port: serverChannel.channel.localAddress!.port!
+            )
+            var serverInboundIterator = serverChannel.inboundStream.makeAsyncIterator()
+            var clientInboundIterator = clientChannel.inboundStream.makeAsyncIterator()
+
+            try await clientChannel.outboundWriter.write("request")
+            try await XCTAsyncAssertEqual(try await serverInboundIterator.next(), "request")
+
+            try await serverChannel.outboundWriter.write("response")
+            try await XCTAsyncAssertEqual(try await clientInboundIterator.next(), "response")
+        }
+    }
+
+    func testDatagramBootstrap_withProtocolNegotiation_andHostPort() throws {
+        XCTAsyncTest {
+            let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 3)
+
+            try await withThrowingTaskGroup(of: NegotiationResult.self) { group in
+                group.addTask {
+                    // We have to use a fixed port here since we only get the channel once protocol negotiation is done
+                    try await self.makeUDPServerChannelWithProtocolNegotiation(
+                        eventLoopGroup: eventLoopGroup,
+                        port: 1995
+                    )
```

Comment:
This is rather interesting. Since we are returning a protocol negotiation result here we can't use random port binding since we can't access the `Channel` before the protocol negotiation is done. I am a bit unsure if this is something we want to support generally since protocol negotiation on a UDP server is weird and we are cheating here in this test
We could change our protocol negotiation methods to instead return an ELF of the result and the channel. WDYT @Lukasa @glbrntt?